semantic-core/src/Control/Effect/Readline.hs

```diff
@@ -93,19 +87,12 @@ runReadlineWithHistory block = do
 
   runReadline prefs settings block
 
--- | Promote a monad transformer into an effect.
-newtype TransC t (m :: * -> *) a = TransC { runTransC :: t m a }
-  deriving (Applicative, Functor, Monad, MonadIO, MonadTrans)
-
-instance (Carrier sig m, Effect sig, Monad (t m), MonadTrans t) => Carrier sig (TransC t m) where
-  eff = TransC . join . lift . eff . handle (pure ()) (pure . (runTransC =<<))
```

Comment:
I realized the other day that `TransC` is actually unnecessary given our intended use of `ReadlineC`, and since I never had any idea if or how it worked (cf the same definition in `path`), I’m happy to 🔥 it.